crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -563,6 +564,11 @@ impl<T: WasmInstance> WasmModuleInstance<T> {
             .with_label_values(&address, reducer_name)
             .observe(timings.total_duration.as_secs_f64());
 
+        // Take a lock on our subscriptions now. Otherwise, we could have a race condition where we commit
+        // the tx, someone adds a subscription and receives this tx as an update, and then receives the
+        // update again when we broadcast_event.
+        let subscriptions = self.info.subscriptions.blocking_read();
```

Comment:
As of right now, we won't have concurrent readers, since reducers execute sequentially. So we probably want this to be a write lock, and use the read lock for the initial subscription, since there we can have concurrent readers.